cmd/secret.go

```diff
@@ -0,0 +1,234 @@
+		}
+		secretValue = string(secretValueBytes)
+	}
+
+	if !encoded {
+		return base64.StdEncoding.EncodeToString([]byte(secretValue)), nil
+	} else {
+		if err := validateProperlyEncoded(secretValue); err != nil {
+			return "", err
+		}
+		return secretValue, nil
+	}
+}
+
+func validateProperlyEncoded(secretValue string) error {
+	if _, err := base64.StdEncoding.DecodeString(secretValue); err != nil {
+		return errors.New("value is not encoded. please use --base64 to let Optimus encode the secret for you.")
+	}
+	return nil
+}
+
+func registerSecret(l log.Logger, conf config.Provider, req *pb.RegisterSecretRequest) (err error) {
+	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
+	defer dialCancel()
+
+	var conn *grpc.ClientConn
+	if conn, err = createConnection(dialTimeoutCtx, conf.GetHost()); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(ErrServerNotReachable(conf.GetHost()).Error())
+		}
+		return err
+	}
+	defer conn.Close()
+
+	secretRequestTimeout, secretRequestCancel := context.WithTimeout(context.Background(), secretTimeout)
+	defer secretRequestCancel()
+
+	l.Info("please wait...")
+	runtime := pb.NewRuntimeServiceClient(conn)
+
+	registerSecretResponse, err := runtime.RegisterSecret(secretRequestTimeout, req)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(coloredError("Secret registration took too long, timing out"))
+		}
+		return errors.Wrapf(err, "Request failed for creating secret %s", req.SecretName)
+	}
+
+	if registerSecretResponse.Success {
+		l.Info(coloredSuccess("Secret registered"))
+	} else {
+		return errors.New(fmt.Sprintf("Request failed for creating secret %s: %s", req.SecretName,
+			registerSecretResponse.Message))
+	}
+
+	return nil
+}
+
+func updateSecret(l log.Logger, conf config.Provider, req *pb.UpdateSecretRequest) (err error) {
+	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
+	defer dialCancel()
+
+	var conn *grpc.ClientConn
+	if conn, err = createConnection(dialTimeoutCtx, conf.GetHost()); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(ErrServerNotReachable(conf.GetHost()).Error())
+		}
+		return err
+	}
+	defer conn.Close()
+
+	secretRequestTimeout, secretRequestCancel := context.WithTimeout(context.Background(), secretTimeout)
+	defer secretRequestCancel()
+
+	l.Info("please wait...")
+	runtime := pb.NewRuntimeServiceClient(conn)
+
+	updateSecretResponse, err := runtime.UpdateSecret(secretRequestTimeout, req)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(coloredError("Secret update took too long, timing out"))
+		}
+		return errors.Wrapf(err, "Request failed for updating secret %s", req.SecretName)
+	}
+
+	if updateSecretResponse.Success {
+		l.Info(coloredSuccess("Secret updated"))
+	} else {
+		return errors.New(fmt.Sprintf("Request failed for updating secret %s: %s", req.SecretName,
```

Comment:
You should not start error with a capital letter as it could be embedded part of some other error. This is done in couple of places in this file

Reply:
Changed the error messages to lowercase